### PR TITLE
Add codecov's PGP public key

### PR DESCRIPTION
**Pull request by paymand**
[keybase.io](https://keybase.io) was down for several hours today and all our CI jobs failed because we failed to download `pgp_keys.asc`.
The purpose of this PR is to bundle the aforementioned file.

One can still request the file to be downloaded from a custom url by setting `BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL`.

### CODE REVIEW

File: hooks/post-command [hunk 1]
```diff
@@ -72,14 +72,22 @@ get_codecov_uploader() {
   fi
 
   # One-time step
-  curl \
-      -fSs \
-      --retry 5 \
-      https://keybase.io/codecovsecurity/pgp_keys.asc \
+  if [[ "${BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL:-false}" = false ]]; then
```
Comment (joscha) [inline hooks/post-command:75]:
```suggestion
  if [[ "${BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL:-false}" = "false" ]]; then
```
To avoid confusion for the reader as false is highlighted as a built-in.

File: hooks/post-command [hunk 2]
```diff
@@ -72,14 +72,22 @@ get_codecov_uploader() {
...
+    cat pgp_keys.asc \
```
Comment (joscha) [inline hooks/post-command:76]: Can we add a path here, do it doesn't matter from where this file is executed, please?

File: pgp_keys.asc [hunk 3]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (joscha) [inline pgp_keys.asc:1]: Can we add a comment to this file as to it's origin and state?

File: plugin.yml [hunk 4]
```diff
@@ -14,6 +14,8 @@ configuration:
       type: boolean
     uploader_version:
       type: string
+    pgp_public_key_url:
+      type: string
```
Comment (joscha) [inline plugin.yml:18]: Can we set this to the keybase url by default and then you can set it to an empty string in your particular set up?

File: hooks/post-command (hunk 1 above)
Comment (paymand) [reply hooks/post-command]: Code changed a bit in the commit I pushed to raise issues: https://github.com/joscha/codecov-buildkite-plugin/pull/28/commits/f69e53c6009e1aedb346796f63387e65559fa35c

File: hooks/post-command (hunk 2 above)
Comment (paymand) [reply hooks/post-command]: Addressed here: https://github.com/joscha/codecov-buildkite-plugin/pull/28/commits/f69e53c6009e1aedb346796f63387e65559fa35c

File: pgp_keys.asc (hunk 3 above)
Comment (paymand) [reply pgp_keys.asc:1]: Addressed here: https://github.com/joscha/codecov-buildkite-plugin/pull/28/commits/f69e53c6009e1aedb346796f63387e65559fa35c

File: plugin.yml (hunk 4 above)
Comment (paymand) [reply plugin.yml:18]: Addressed here: https://github.com/joscha/codecov-buildkite-plugin/pull/28/commits/f69e53c6009e1aedb346796f63387e65559fa35c

**Comment by paymand** [comment]
@joscha pushed a commit (and a fixup) to address the raised issues.

File: hooks/post-command (hunk 2 above)
Comment (paymand) [reply hooks/post-command]: I added `$(pwd)`. Not sure if that cuts it, please let me know if you have a better suggestion though.

File: hooks/post-command [hunk 5]
```diff
@@ -72,14 +72,26 @@ get_codecov_uploader() {
   fi
 
   # One-time step
-  curl \
-      -fSs \
-      --retry 5 \
-      https://keybase.io/codecovsecurity/pgp_keys.asc \
+  if [[ "${BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL-"false"}" = "" ]]; then
+    debug "Importing bundled Codecov's PGP public key : $(pwd)/pgp_keys.asc" 
+    cat $(pwd)/pgp_keys.asc \
```
Comment (joscha) [inline hooks/post-command:77]:
```suggestion
    cat "$(pwd)/pgp_keys.asc" \
```

due to https://www.shellcheck.net/wiki/SC2046

File: hooks/post-command [hunk 6]
```diff
@@ -72,14 +72,26 @@ get_codecov_uploader() {
   fi
 
   # One-time step
-  curl \
-      -fSs \
-      --retry 5 \
-      https://keybase.io/codecovsecurity/pgp_keys.asc \
+  if [[ "${BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL-"false"}" = "" ]]; then
```
[the remark this answers is not in the document]
Comment (paymand) [reply hooks/post-command:75]: so we wanted to achieve this behavior:
1. If `BUILDKITE_PLUGIN_CODECOV_PGP_PUBLIC_KEY_URL` is set to empty string, i.e. `""` => use the bundled PGP public key.
2. If is not set => download from keybase.io
3. If set to a url => download from that url.
Based on [docs](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion):
"Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both parameter’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence."

File: hooks/post-command (hunk 5 above)
Comment (paymand) [reply hooks/post-command]: Fixed in https://github.com/joscha/codecov-buildkite-plugin/pull/28/commits/e62af2b9ad26c4851b5964adf96417dd74f2e0a8

**Comment by paymand** [comment]
@joscha sorry for the late response. Addressed one raised issue and left a comment for the other.
